### ISSUE TITLE
Bug: [@typescript-eslint/no-useless-constructor] the rule doesn't work on public constructors

### ISSUE THREAD

**Issue by ODGodinho** [typescript-eslint/typescript-eslint#6094]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&showAST=es&code=LAKAxgNghgzjAEAhWBTeBvUp7wA4FcAjCASzHgDsoBbFALnhgBcAnEigcwG4sQcDiZeGAD2FZi3xgmIlgAoAlBmw54TABYkYAOiq14AXngAiZDBTGefeAF9edkKBQAPXLKbDoceABUUzAGEvBBcmFAoAEwQzNExHawB6BPgAOREPFhQ3Fg8AVQoIfxgVAVJyUXFWKRl5JTjVRnxcFFqrHAdQDviXbI9IWAQ-QOCAJnhQ8KikVGV4nCT4AHkAaXh8AqKVColq2UVZhpgmlsU223sgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS0yHJsnxwy+DtHRgA2uGw5E0aNMiKIAXUUBfELqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
Example dont report unless-constructor


class Base {

    public name: string;

    public constructor() {
        this.name = "test";
    }

}

export class TestClass extends Base {
    
    // This should report unless-constructor
    public constructor() {
        super();
    }

}
```
```


### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 2022,
    sourceType: "module",
    project: [ "tsconfig.json" ], // Specify it only for TypeScript files
  },
  rules: {
    "no-useless-constructor": [ "off" ],
    "@typescript-eslint/no-useless-constructor": [ "error" ],
  },
};
```

### tsconfig

```jsonc
{
    "compilerOptions": {
        "baseUrl": ".",
        "outDir": "./dist",
        "rootDirs": [
            ".",
            "./src"
        ],
        "typeRoots": [
            "./src/@types",
            "node_modules/@types"
        ],
        "types": [ "node" ]
    },
    "exclude": [ "./coverage/" ],
    "extends": "@odg/tsconfig/tsconfig.node.json",
    "include": [ "." ]
}
```

### Expected Result

 // report unless expected
```typescript
    public constructor() {
        super();
    }
 ```

 // Ok Reporting
```typescript
    constructor() {
        super();
    }
 ```

### Actual Result

- public constructor don't report unless
 // Don't working (no report)
```typescript
    public constructor() {
        super();
    }
 ```

### Additional Info

 yarn eslint --debug .\src\test.ts                                                              
yarn run v1.22.19
$ F:\projetos\ODG-Linter-Js\node_modules\.bin\eslint --debug .\src\test.ts
  eslint:cli CLI args: [ '--debug', '.\\src\\test.ts' ] +0ms
  eslint:cli Using flat config? false +2ms
  eslint:cli Running on files +3ms
  eslintrc:config-array-factory Loading JSON config file: F:\projetos\ODG-Linter-Js\package.json +0ms
  eslintrc:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: 'F:\\projetos\\ODG-Linter-Js', loose: false } ] +0ms
  eslintrc:ignore-pattern   processed: { basePath: 'F:\\projetos\\ODG-Linter-Js', patterns: [ '/**/node_modules/*' ] } +1ms
  eslintrc:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: 'F:\\projetos\\ODG-Linter-Js', loose: false } ] +1ms
  eslintrc:ignore-pattern   processed: { basePath: 'F:\\projetos\\ODG-Linter-Js', patterns: [ '/**/node_modules/*' ] } +0ms
  eslint:file-enumerator Start to iterate files: [ '.\\src\\test.ts' ] +0ms
  eslint:file-enumerator File: F:\projetos\ODG-Linter-Js\src\test.ts +0ms
  eslintrc:cascading-config-array-factory Load config files for F:\projetos\ODG-Linter-Js\src. +0ms
  eslintrc:cascading-config-array-factory No cache found: F:\projetos\ODG-Linter-Js\src. +1ms
  eslintrc:config-array-factory Config file not found on F:\projetos\ODG-Linter-Js\src +4ms
  eslintrc:cascading-config-array-factory No cache found: F:\projetos\ODG-Linter-Js. +0ms
  eslintrc:config-array-factory Loading legacy config file: F:\projetos\ODG-Linter-Js\.eslintrc +1ms
  eslintrc:config-array-factory Config file found: F:\projetos\ODG-Linter-Js\.eslintrc +12ms
  eslintrc:config-array-factory Loading parser "@typescript-eslint/parser" from F:\projetos\ODG-Linter-Js\.eslintrc +0ms
  eslintrc:config-array-factory Loaded: @typescript-eslint/parser@5.43.0 (F:\projetos\ODG-Linter-Js\node_modules\@typescript-eslint\parser\dist\index.js) +1ms
  eslintrc:config-array-factory Loading plugin "@typescript-eslint" from F:\projetos\ODG-Linter-Js\.eslintrc +255ms
  eslintrc:config-array-factory Loaded: @typescript-eslint/eslint-plugin@5.43.0 (F:\projetos\ODG-Linter-Js\node_modules\@typescript-eslint\eslint-plugin\dist\index.js) +1ms
  eslint:rules Loading rule 'brace-style' (remaining=289) +0ms
  eslint:rules Loading rule 'comma-dangle' (remaining=288) +4ms
  eslint:rules Loading rule 'dot-notation' (remaining=287) +7ms
  eslint:rules Loading rule 'indent' (remaining=286) +4ms
  eslint:rules Loading rule 'init-declarations' (remaining=285) +8ms
  eslint:rules Loading rule 'keyword-spacing' (remaining=284) +1ms
  eslint:rules Loading rule 'lines-between-class-members' (remaining=283) +1ms
  eslint:rules Loading rule 'no-dupe-class-members' (remaining=282) +13ms
  eslint:rules Loading rule 'no-duplicate-imports' (remaining=281) +2ms
  eslint:rules Loading rule 'no-empty-function' (remaining=280) +1ms
  eslint:rules Loading rule 'no-extra-parens' (remaining=279) +2ms
  eslint:rules Loading rule 'no-extra-semi' (remaining=278) +2ms
  eslint:rules Loading rule 'no-invalid-this' (remaining=277) +6ms
  eslint:rules Loading rule 'no-loop-func' (remaining=276) +2ms
  eslint:rules Loading rule 'no-loss-of-precision' (remaining=275) +1ms
  eslint:rules Loading rule 'no-magic-numbers' (remaining=274) +1ms
  eslint:rules Loading rule 'no-restricted-imports' (remaining=273) +13ms
  eslint:rules Loading rule 'no-unused-expressions' (remaining=272) +18ms
  eslint:rules Loading rule 'no-useless-constructor' (remaining=271) +2ms
  eslint:rules Loading rule 'object-curly-spacing' (remaining=270) +6ms
  eslint:rules Loading rule 'quotes' (remaining=269) +17ms
  eslint:rules Loading rule 'semi' (remaining=268) +7ms
  eslint:rules Loading rule 'space-before-blocks' (remaining=267) +2ms
  eslint:rules Loading rule 'space-infix-ops' (remaining=266) +1ms
  eslintrc:config-array-factory Plugin F:\projetos\ODG-Linter-Js\node_modules\@typescript-eslint\eslint-plugin\dist\index.js loaded in: 196ms +196ms
  eslintrc:cascading-config-array-factory No cache found: F:\projetos. +466ms
  eslintrc:config-array-factory Config file not found on F:\projetos +1ms
  eslintrc:cascading-config-array-factory No cache found: F:\. +1ms
  eslintrc:config-array-factory Config file not found on F:\ +0ms
  eslintrc:cascading-config-array-factory Configuration was determined: ConfigArray(2) [ { type: 'config', name: 'DefaultIgnorePattern', filePath: '', criteria: null, env: undefined, globals: undefined, ignorePattern: IgnorePattern { patterns: [Array], basePath: 'F:\\projetos\\ODG-Linter-Js', loose: false }, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { type: 'config', name: '.eslintrc', filePath: 'F:\\projetos\\ODG-Linter-Js\\.eslintrc', criteria: null, env: { browser: true, es2021: true, jest: true, node: true }, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: { error: null, filePath: 'F:\\projetos\\ODG-Linter-Js\\node_modules\\@typescript-eslint\\parser\\dist\\index.js', id: '@typescript-eslint/parser', importerName: '.eslintrc', importerPath: 'F:\\projetos\\ODG-Linter-Js\\.eslintrc' }, parserOptions: { ecmaFeatures: [Object], ecmaVersion: 2022, sourceType: 'module', project: [Array] }, plugins: { '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/no-useless-constructor': [Array] }, settings: undefined } ] on F:\projetos\ODG-Linter-Js\src +1ms
  eslintrc:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: 'F:\\projetos\\ODG-Linter-Js', loose: false } ] +472ms
  eslintrc:ignore-pattern   processed: { basePath: 'F:\\projetos\\ODG-Linter-Js', patterns: [ '/**/node_modules/*' ] } +0ms
  eslintrc:ignore-pattern Check {
  filePath: 'F:\\projetos\\ODG-Linter-Js\\src\\test.ts',
  dot: false,
  relativePath: 'src/test.ts',
  result: false
} +1ms
  eslint:cli-engine Lint F:\projetos\ODG-Linter-Js\src\test.ts +0ms
  eslint:linter Linting code for F:\projetos\ODG-Linter-Js\src\test.ts (pass 1) +0ms
  eslint:linter Verify +1ms
  eslint:linter With ConfigArray: F:\projetos\ODG-Linter-Js\src\test.ts +0ms
  eslint:linter Parsing: F:\projetos\ODG-Linter-Js\src\test.ts +1ms
  eslint:linter Parsing successful: F:\projetos\ODG-Linter-Js\src\test.ts +645ms
  eslint:linter Scope analysis: F:\projetos\ODG-Linter-Js\src\test.ts +1ms
  eslint:linter Scope analysis successful: F:\projetos\ODG-Linter-Js\src\test.ts +0ms
  eslint:config-comment-parser Parsing list config +0ms
  eslint:rules Loading rule 'max-classes-per-file' (remaining=265) +665ms
  eslint:linter Generating fixed text for F:\projetos\ODG-Linter-Js\src\test.ts (pass 1) +7ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:file-enumerator Complete iterating files: [".\\src\\test.ts"] +1s
  eslint:cli-engine Linting complete in: 1129ms +656ms
Done in 1.40s.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.43.0` |
| `@typescript-eslint/parser`            | `5.43.0` |
| `TypeScript`                           | `4.8.4` |
| `ESLint`                               | `8.27.0` |
| `node`                                 | `16.16.0` |

**Comment by bradzacher** [comment]
This is intentional https://github.com/typescript-eslint/typescript-eslint/issues/948
Please use the search for opened and closed issues!
